File: ruby/leetcode001.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum( nums, target )
    table = {}
    nums.each_with_index { |e, i|
        return [table[e] + 1, i + 1] if table.has_key?(e)
        table[target - e] = i
    }
end

File: Archive#00-001-100/001-Two Sum/leetcode001.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        table = {}
        for i, e in enumerate(nums):
            if table.get(e) != None:
                return [table[e], i]
            table[target - e] = i
